apps/E2E/src/common/BasePage.ts

```diff
@@ -209,33 +213,86 @@ export abstract class BasePage {
     });
   }
 
-  /* Scrolls to the specified or primary UI test element until it is displayed. */
-  async scrollToTestElement(component?: WebdriverIO.Element): Promise<void> {
-    const ComponentToScrollTo = component ?? (await this._primaryComponent);
-    if (await ComponentToScrollTo.isDisplayed()) {
+  /* Scrolls to the specified or primary UI test element until it is displayed.
+   * Either the component identifier or the component itself can be passed in as a parameter.
+   * On Android only the component identifier can be passed in.
+   * If no parameter is passed in, the primary component is used.
+   */
+  async scrollToTestElement(component?: WebdriverIO.Element | string): Promise<void> {
+    let componentIdentifier: string;
+    let componentToScrollTo: WebdriverIO.Element;
+
+    if (typeof component == 'undefined') {
+      componentIdentifier = this._primaryComponentName;
+      componentToScrollTo = await this._primaryComponent;
+    } else if (typeof component == 'string') {
+      componentIdentifier = component;
+      componentToScrollTo = await By(componentIdentifier);
+    } else {
+      componentIdentifier = null;
```

Comment:
Won't this cause an issue with line 262?

Reply:
On Android only string would be supported.
If the webdriverio.element is retrievable to be passed into the scroll function, then the element is already visible, no need to scroll, hence line 236 would cause the function to return prematurely, never reaching line 262